MAINT, DOC: add index for user docs.

rossbar opened the pull request:
This PR represents one option to resolve #16372 

Re-adds an index.rst for the user documentation to serve as a start page for the latex/pdf version of the user documentation. The index.rst is taken verbatim from the source prior to the doc restructuring per NEP 44 (see 22356fc e.g.). The index is now marked as an orphan, so it should not interfere with the build/display of the html documentation, though this means that the page *will* be visible if someone manually navigates to `docs/user`.

rossbar committed:
MAINT,DOC: add index for user docs.

Adds an index.rst for the user documentation (pre-NEP 44) to
serve as a start page for the latex version of the user
documentation.

Marked as an orphan, so should not interfere with building html
docs

rossbar commented:
A solution to #16372 will be necessary to complete the documentation distribution step for the 1.19 release, so I will label this as a backport candidate.

melissawm commented on doc/source/user/index.rst:
```diff
@@ -0,0 +1,23 @@
+   quickstart
+   basics
+   misc
+   numpy-for-matlab-users
```
I'm wondering if we should include a link to `tutorials_index.rst` here to simulate the `For users` section of the main docs page

rossbar replied:
Good suggestion - the `howtos_index` and `absolute_beginners` are the two other documents that are missing from the latex user guide as well. I will add all three.

melissawm commented:
Makes sense to me, thanks @rossbar !

rossbar committed:
Update userguide index.

Adds the following to the LaTeX/pdf docs:
 * Absolute beginners tutorial
 * Tutorials
 * HowTos

rossbar committed:
Re-order user/index so indices are last.

Having the \*_index listings higher up in the user/index clobbers
the remainder of the content in the toctree. Simplest solution is
to just move them to last. Alternatively, could have multiple
toctrees with different depths.

Qiyu8 commented on doc/source/user/index.rst:
```diff
@@ -0,0 +1,26 @@
+   numpy-for-matlab-users
+   building
+   c-info
+   tutorials_index
```
this document should be tutorial-svd?  because tutorials_index is almost the same with index.rst.

melissawm replied:
So, maybe this should be remapped.  `index.rst` should contain everything under "For users" in this page: https://numpy.org/devdocs/

`tutorials_index.rst` should be just the tutorials (although some of them might show up twice because the quickstart tutorial and the absolute beginners tutorial show up at the main index page).

rossbar replied:
> this document should be tutorial-svd? because tutorials_index is almost the same with index.rst.

The reason for including the index rather than the tutorials individually is that there will be more tutorials in the future, and this should capture them automatically. Sphinx is smart enough to non replicate redundant entries in the toc, so the quickstart, absolute beginner, etc. tutorials are not duplicated (also, see below about user guide layout).

> So, maybe this should be remapped. index.rst should contain everything under "For users" in this page: https://numpy.org/devdocs/

In the current layout (with the *_index files included, but the original toc entries not removed) each entry gets it's own chapter in the user guide. If we remove things like `basics`, `c-info`, `building` etc. from the toc here in favor of the `tutorials_index`, then these entries become sections in a larger chapter called "tutorials". In other words, the only thing affected on the `latex` side of things is the layout of the user guide: it's a matter of preference whether we have individual chapters for "miscellaneous", "using the c-api", etc., or one big chapter with these as sections.

Since the content is the same, I don't mind either way - if you prefer the "big chapter" approach over the current layout, LMK and I will modify it!

rossbar replied:
On second thought, this might be a nice place for the `.. only` directive: we can use one toctree to explicitly control chapter layout for the latex-version of the user guide, and a different one for the html output that matches the layout of the "For users" section in the html docs.

mattip replied:
I am pretty sure you should avoid the rabbit-hole of `.. only` since I do not think it works very well.

melissawm replied:
I think this is fine as is, if needs change in the near future we can always open a new issue. Thanks, @rossbar :)

mattip commented:
Thanks @rossbar. Putting this in to unblock building the pdf documentation. Please revisit as needed to tweek it.
